### PR TITLE
Lazily initialize decks of cards instead of initializing thousands of cards at once (.09s drop in init times)

### PR DESCRIPTION
**Pull request by Mothblocks**
We're getting into the less than 0.1s realm of low hanging fruits now.

Decks of cards were initializing thousands of cards at once, not unlike [paper bins](https://github.com/tgstation/tgstation/pull/69586) or [circuit components](https://github.com/tgstation/tgstation/pull/69664). 

Cards are more complicated than paper bins since we can't just keep a count, and managing individual card object lifetimes is significantly more complicated than incrementing/decrementing a number. Thus, the logic here is slightly different.

Decks now have an `initial_cards` variable which can take card names (which is most implementers), and for special decks, can specify an interface that is basically a lazy function for creating a card atom.

When anything needs any real card atom, we just generate them all. The runtime cost of this is extremely small, and affects neither dev cycles (the motivation for the change) or ongoing rounds.

### CODE REVIEW

File: code/modules/cards/cardhand.dm [hunk 1]
```diff
@@ -6,26 +6,9 @@
...
-		for(var/card_name in cards)
-			var/obj/item/toy/singlecard/new_card = new (loc, card_name)
-			new_card.update_appearance()
-			cards_to_combine += new_card
-		cards = list() // reset our cards to an empty list
-
-	for(var/obj/item/toy/singlecard/new_card in cards_to_combine)
-		new_card.forceMove(src)
-		cards += new_card
-
```
Comment (Mothblocks) [inline code/modules/cards/cardhand.dm:28]: Unused

Comment (timothymtorres) [reply code/modules/cards/cardhand.dm:28]: https://github.com/tgstation/tgstation/blob/f4af90780aa1f8e6855945bde76bb82a862d7815/code/modules/cards/singlecard.dm#L171

Comment (Mothblocks) [reply code/modules/cards/cardhand.dm:28]: Ugh

Comment (Mothblocks) [reply code/modules/cards/cardhand.dm:28]: What if I just make that:
```
		var/obj/item/toy/cards/cardhand/new_cardhand = new (drop_location())
		new_cardhand.insert(card)
```

Comment (Mothblocks) [reply code/modules/cards/cardhand.dm:28]: Ok it doesn't work but I'll figure it out

**Review by timothymtorres — APPROVED** [review]
dreamseeker_acGXANGTax

Looks good!  Did some testing and found some bugs that I listed in #71125, but they are unrelated to your changes.  Everything else appears to work normally.

**Comment by timothymtorres** [comment]
Missed a spot in:

https://github.com/tgstation/tgstation/blob/8f5e17163fab9d1284c57202af299f85c0b9ace8/code/modules/holodeck/items.dm#L35-L51

**Comment by Mothblocks** [comment]
O_o the var doesn't exist anymore, it was renamed so stuff like that would be seen in compile

let me look

**Comment by Mothblocks** [comment]
You're looking in the wrong place or something. It's fine for me:

```dm
/obj/item/toy/cards/deck/syndicate/holographic/proc/handle_card_delete(datum/source)
	SIGNAL_HANDLER

	//if any REAL cards have been inserted into the deck they are moved outside before destroying it
	for(var/obj/item/toy/singlecard/card in card_atoms)
		if(card.flags_1 & HOLOGRAM_1)
			continue
		card_atoms -= card
		card.forceMove(drop_location())

```

**Comment by timothymtorres** [comment]
You are right, maybe I was looking at the wrong branch when I was doing a final glance at the code.  Everything looks fine.

**Pull request merged**